### ISSUE TITLE
Slow cell execution time after plotting images

### ISSUE THREAD

**Issue by ido-greenfeld** [microsoft/vscode#143232]
## Environment data

-   VS Code version: 1.64.2
-   Jupyter Extension version: v2022.0.1814523869
-   Python Extension version: v2022.0.1814523869
-   Local OS - Windows 10
-   Remote OS - Ubuntu 20.04.2
-   Python version - 3.8.10
-   Jupyter server running: Remove VM on GCP

## Actual behaviour

I'm running a remote VM on GCP and I'm using remote-ssh extension.

After plotting some images, cell executions become really slow.
Sometimes it takes a few second between running a cell (shift + enter) and seeing the execution stopwatch.
In other cases the execution stopwatch takes a really long time (10 second for easy lines like `a = 1` and once it's finished it jumps back to a short time like 1 second).
This slowdown also happens when running empty cells which leads me to suspect this is a connection issue to the VM... however, if I don't plot any images, everything runs smoothly.

When I run the same notebook on the same environment in Jupyterlab it runs as it should.

## Steps to reproduce:

Tried the following code for example:
``` 
import numpy as np
import matplotlib.pyplot as plt

for i in range(20):
    rand_im = np.random.rand(600, 600)
    fig = plt.figure(figsize=(5,5))
    plt.imshow(rand_im)
```

In the gif bellow you can see the behavior I'm experiencing - 
- I first run the plotting loop. The stopwatch reaches ~10 secs and then jumps back to 4 sec.
- I then run an import which doesn't respond at first for a few seconds and then starts running.
- After that the `a=1` cell runs without issue.
- And then the `b=2` again freezes and even the cell after it which is empty takes time to run.

![● agrobrain-segmentation-training  SSH_ ido-base-dl-2  - test ipynb 2022-02-14 17-02-24 (1)](https://user-images.githubusercontent.com/53991912/153891837-33a7a98a-5ad6-4272-bf72-8313aa4bcfb3.gif)

## Logs

Output from the Jupyter output panel - 

```
Info 15:02:04: Got new session c0d0ab06-961c-49ac-9683-c5830435390a
Info 15:02:04: Started new restart session
Info 15:02:04: Dispose Kernel process
Info 15:02:04: Executing silently Code (idle) = import sys\nprint(sys.executable)
Info 15:02:04: Executing silently Code (idle) = import sys\nprint(sys.executable)
Info 15:02:04: Executing silently Code (idle) = import sys\nprint(sys.executable)
Info 15:02:04: Executing silently Code (completed) = import sys\nprint(sys.executable)
Info 15:02:04: Executing silently Code (completed) = import sys\nprint(sys.executable)
Info 15:02:04: Executing silently Code (completed) = import sys\nprint(sys.executable)
Info 15:02:08: UpdateWorkingDirectoryAndPath in Kernel
Info 15:02:08: UpdateWorkingDirectoryAndPath in Kernel
Info 15:02:08: UpdateWorkingDirectoryAndPath in Kernel
Info 15:02:08: Executing silently Code (idle) = import site\nsite.addsitedir(site.getusersitepackages())\nimport os\nimport sys\n%cd "/home/ido_taranis_
Info 15:02:08: Executing silently Code (idle) = import site\nsite.addsitedir(site.getusersitepackages())\nimport os\nimport sys\n%cd "/home/ido_taranis_
Info 15:02:08: Executing silently Code (idle) = import site\nsite.addsitedir(site.getusersitepackages())\nimport os\nimport sys\n%cd "/home/ido_taranis_
Info 15:02:08: Executing silently Code (completed) = import site\nsite.addsitedir(site.getusersitepackages())\nimport os\nimport sys\n%cd "/home/ido_taranis_
Info 15:02:08: Waiting for idle on (kernel): c0d0ab06-961c-49ac-9683-c5830435390a -> idle
Info 15:02:08: Finished waiting for idle on (kernel): c0d0ab06-961c-49ac-9683-c5830435390a -> idle
Info 15:02:08: Executing silently Code (completed) = import site\nsite.addsitedir(site.getusersitepackages())\nimport os\nimport sys\n%cd "/home/ido_taranis_
Info 15:02:08: Waiting for idle on (kernel): c0d0ab06-961c-49ac-9683-c5830435390a -> idle
Info 15:02:08: Finished waiting for idle on (kernel): c0d0ab06-961c-49ac-9683-c5830435390a -> idle
Info 15:02:08: Executing silently Code (completed) = import site\nsite.addsitedir(site.getusersitepackages())\nimport os\nimport sys\n%cd "/home/ido_taranis_
Info 15:02:08: Waiting for idle on (kernel): c0d0ab06-961c-49ac-9683-c5830435390a -> idle
Info 15:02:08: Finished waiting for idle on (kernel): c0d0ab06-961c-49ac-9683-c5830435390a -> idle
Info 15:02:26: Execute Cells request 0
Info 15:02:26: Execute Cell 0 ~/repositories/agrobrain-segmentation-training/agrobrain_segmentation_training/notebooks/data_manipulation/test.ipynb
Info 15:02:30: Cell 0 executed with state Success
Info 15:02:49: Execute Cells request 1
Info 15:02:49: Execute Cell 1 ~/repositories/agrobrain-segmentation-training/agrobrain_segmentation_training/notebooks/data_manipulation/test.ipynb
Info 15:02:52: Cell 1 executed with state Success
Info 15:02:56: Execute Cells request 2
Info 15:02:56: Execute Cell 2 ~/repositories/agrobrain-segmentation-training/agrobrain_segmentation_training/notebooks/data_manipulation/test.ipynb
Info 15:02:57: Cell 2 executed with state Success
Info 15:03:02: Execute Cells request 3
Info 15:03:02: Execute Cell 3 ~/repositories/agrobrain-segmentation-training/agrobrain_segmentation_training/notebooks/data_manipulation/test.ipynb
Info 15:03:05: Cell 3 executed with state Success
Info 15:03:06: Execute Cells request 4
Info 15:03:06: Execute Cell 4 ~/repositories/agrobrain-segmentation-training/agrobrain_segmentation_training/notebooks/data_manipulation/test.ipynb
Info 15:03:10: Cell 4 executed with state Idle
```

**Comment by ido-greenfeld** [comment]
Hi @DonJayamanne, I see you moved the issue from vscode-jupyter to vscode...
Is this not jupyter related?

**Comment by DonJayamanne** [comment]
Its specific to remote SSH, we've found in the past that such delays are caused by VS Code when running in remote SSH.

**Comment by ido-greenfeld** [comment]
I see...
I think I can confirm that this is indeed the case and the fault is not with the jupyter extension.

I tried running in a different way, these are the step I took - 
1. Connected to my VM using SSH, not using the remote-SSH extension but doing it directly - I'm on GCP so I used the `gcloud compute ssh` command
2. Started the jupyter server on my VM and port forwarded to my local machine (this is how I used to work before using vscode, by opening jupyter lab in the browser after port forwarding)
3. Opened vscode and used the jupyter extension to connect to the jupyter server - `Jupyter: specify local or remote...`
4. Changed kernel to use the jupyter server kernel

So now I'm connected to my jupyter server without going through the remote-ssh extension.
When running the code I wrote above I get no delays whatsoever... which indicated that the issue is indeed with the different way of connecting using remote-ssh
